sdk/src/calibration.cpp

```diff
@@ -139,6 +139,39 @@ aditof::Status Calibration::getGainOffset(const std::string &mode, float &gain,
     return Status::GENERIC_ERROR;
 }
 
+aditof::Status Calibration::getIntrinsic(float key,
+										 std::vector<float> &data) const {
+    using namespace aditof;
+
+	bool validParam = (INTRINSIC == key) || (DISTORTION_COEFFICIENTS == key);
+
+	if (!validParam) {
+			LOG(WARNING) << "Invalid intrinsic " << std::to_string(key).c_str();
+			return Status::INVALID_ARGUMENT;
+	}
+
+    for (const auto &mapElement : m_calibration_map) {
+        float mapKey = mapElement.first;
+        const packet_struct &sub_packet_map = mapElement.second;
+
+        if (mapKey == CAMERA_INTRINSIC) {
+            for (const auto &packet : sub_packet_map.packet) {
+							float packetKey = packet.first;
+
+							if (packetKey == key) {
+								const param_struct &param_map = packet.second;
+								const std::list<float> &valList = param_map.value;
+
+								data.insert(data.begin(), valList.begin(), valList.end());
+								return Status::OK;
+							}
+            }
+        }
+    }
+
```

Comment:
You could add a warning message that indicates that there are no intrinsic parameters stored in the device memory for the given key.

Reply:
solved in commit https://github.com/analogdevicesinc/aditof_sdk/pull/203/commits/adba4901642e22b8626a1910e3c23271c2af6dc3